Server/BackEndMonitoringServer/src/CLogicalDiskStatus.cpp

```diff
@@ -0,0 +1,83 @@
+#include "stdafx.h"
+
+#include "CLogicalDiskStatus.h"
+
+CLogicalDiskStatus* CLogicalDiskStatus::FactoryLogicalDiskStatus(
+	const std::string& disk_name,
+	EMemoryCountType count_type)
+{
+	CLogicalDiskStatus* p_created_object = new CLogicalDiskStatus(
+		disk_name, count_type);
+
+	if (!p_created_object->TryUpdateCurrentStatus())
+	{
+		return nullptr;
+	}
+
+	return p_created_object;
+}
+
+bool CLogicalDiskStatus::TryUpdateCurrentStatus()
+{
+	//try-catch block for avoidind floppy disks
+	try
+	{
+		m_disk_info = std::filesystem::space(m_disk_name);
+	}
+	catch (std::filesystem::filesystem_error&)
+	{
+		return false;
+	}
+	if (m_disk_info.capacity == NULL &&
```

Comment:
nullptr

Reply:
If filesystem::space wasn't able to find disk - it's return object wih ALL empty fields. That's why I need to check all of them for NULL